frontend/src/components/auth/auth.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect, useContext } from 'react'
+import { auth } from '../../../../backend/firebase/firebase'
+import { onAuthStateChanged } from 'firebase/auth'
+
+const AuthContext = React.createContext()
+
+export function useAuth() {
+    return useContext(AuthContext)
+}
+
+export function AuthProvider({ children }) {
```

Comment:
Can we add some comments here on what these functions are doing? Are there some TODOs we need to add here related to setting up your user database?